core/src/main/java/org/elasticsearch/http/HttpTransportSettings.java

```diff
@@ -69,7 +70,14 @@
     public static final Setting<Boolean> SETTING_HTTP_DETAILED_ERRORS_ENABLED =
         Setting.boolSetting("http.detailed_errors.enabled", true, Property.NodeScope);
     public static final Setting<Boolean> SETTING_HTTP_CONTENT_TYPE_REQUIRED =
-        Setting.boolSetting("http.content_type.required", false, Property.NodeScope);
+        new Setting<>("http.content_type.required", (s) -> Boolean.toString(true), (s) -> {
+            final boolean value = Booleans.parseBoolean(s);
+            if (value == false) {
+                throw new IllegalArgumentException("http.content_type.required cannot be set to false. It exists only to make a rolling" +
+                    " upgrade easier");
+            }
+            return true;
+        }, Property.NodeScope, Property.Deprecated);
```

Comment:
do we have a test for this? just making sure that a 5.x cluster having the setting set to false can upgrade to 6.0.